data/virt_autotest/virt_logs_collector.sh

```diff
@@ -0,0 +1,427 @@
+      exit 1
+      ;;
+    *)
+      help_usage | tee -a ${virt_logs_collecor_log}
+      exit 1
+      ;;
+  esac
+done
+shift "$(($OPTIND -1))"
+if [[ ${virt_logs_folder} == "" ]];then
+   virt_logs_folder="/tmp/virt_logs_residence"
+fi
+rm -f -r ${virt_logs_folder}
+if [[ ${virt_guests_wanted} == "" ]];then
+   virt_guests_wanted="all"
+fi
+
+unset guest_hash_ipaddr
+declare -a guest_hash_ipaddr=""
+guest_domain_types="sles"
+guests_inactive_array=`virsh list --inactive | grep -E "${guest_domain_types}" | awk '{print $2}'`
+guest_domains_array=`virsh list  --all | grep -E "${guest_domain_types}" | awk '{print $2}'`
+guest_macaddresses_array=""
+guest_ipaddress="";
+guest_hash_index=0
+guest_current=""
+dhcpd_lease_file="/var/lib/dhcp/db/dhcpd.leases"
+
+#Establish virtual machine domain name and ip address mapping
+for guest_current in ${guest_domains_array[@]};do
+    guest_macaddresses_array[${guest_hash_index}]=`virsh domiflist --domain ${guest_current} | grep -oE "([0-9|a-z]{2}:){5}[0-9|a-z]{2}"`
+    guest_ipaddress=`tac $dhcpd_lease_file | awk '!($0 in S) {print; S[$0]}' | tac | grep -iE "${guest_macaddresses_array[${guest_hash_index}]}" -B8 | grep -oE "([0-9]{1,3}\.){3}[0-9]{1,3}" | tail -1`
+    guest_hash_ipaddr[${guest_hash_index}]=${guest_ipaddress}
+    echo -e ${guest_current}:${guest_hash_ipaddr[${guest_hash_index}]} | tee -a ${virt_logs_collecor_log}
+    guest_hash_index=$(( ${guest_hash_index} + 1 ))
+done
+
+#Start collecing logs from host and virtual machine
+setup_common_logs_folder ${virt_logs_folder}	
+echo -e "collect_supportconfig ${virt_logs_folder} host | tee -a ${virt_logs_collecor_log}"
+collect_supportconfig ${virt_logs_folder} host | tee -a ${virt_logs_collecor_log}
+virt_logs_collector_result=$(( ${virt_logs_collector_result} | $? ))
+echo -e "collect_extra_logs_from_host ${virt_logs_folder} ${virt_extra_logs_host} | tee -a ${virt_logs_collecor_log}"
+collect_extra_logs_from_host ${virt_logs_folder} ${virt_extra_logs_host} | tee -a ${virt_logs_collecor_log}
+virt_logs_collector_result=$(( ${virt_logs_collector_result} | $? ))
+if [[ ${virt_guests_wanted} == "none" ]];then
+   echo -e "Will not collect supportconfig from any guest.\n" | tee -a ${virt_logs_collecor_log}
+else
+   guest_hash_index=0
+   for guest_current in ${guest_domains_array[@]};do
+       if [[ ${virt_guests_wanted} == "all" ]] || [[ ${virt_guests_wanted} =~ .*${guest_current}.* ]];then
+          if [[ ${guests_inactive_array[@]} =~ .*${guest_current}.* ]];then 
+             echo -e "Virtual machine ${guest_current} in shutdown state. Skip collecting logs from it." | tee -a ${virt_logs_collecor_log}
+	  else
+             echo -e "collect_supportconfig ${virt_logs_folder} guest ${guest_hash_ipaddr[${guest_hash_index}]} ${guest_current}" | tee -a ${virt_logs_collecor_log}
+             collect_supportconfig ${virt_logs_folder} guest ${guest_hash_ipaddr[${guest_hash_index}]} ${guest_current} | tee -a ${virt_logs_collecor_log}
```

Comment:
After collecting guest's supportconfig, it seems from code that the logs are in guest's specified folder. I guess there will be some code in openqa to upload them from the guest to openqa, right?

Reply:
There is another pull request to fetch collected logs from guest to host. Then openQA can handle them only on host.